tests/functional/compile/test_compile.py

```diff
@@ -187,12 +187,23 @@ def test_compile_inline_not_add_node(self, project):
         manifest = parse_result.result
         assert len(manifest.nodes) == 4
         dbt = dbtRunner(manifest=manifest)
-        dbt.invoke(
-            ["compile", "--inline", "select * from {{ ref('second_model') }}"],
-            populate_cache=False,
-        )
+        dbt = dbtRunner()
         assert len(manifest.nodes) == 4
 
+    def test_compile_inline_syntax_error(self, project, mocker):
+        patched_fire_event = mocker.patch("dbt.task.compile.fire_event")
+        dbt = dbtRunner()
+        res = dbt.invoke(["compile", "--inline", "select * from {{ ref(1) }}"])
+        patched_fire_event.assert_called_once()
```

Comment:
I didn't really find a better way to test things here other than making sure we call fire event and re-raised a proper error.